service/frontend/accessControlledHandler.go

```diff
@@ -121,11 +121,15 @@ func (a *AccessControlledWorkflowHandler) DeprecateDomain(
 	request *shared.DeprecateDomainRequest,
 ) error {
 
+	scope := a.GetResource().GetMetricsClient().
```

Comment:
Can we use `getMetricsScopeWithDomain` here?

Reply:
Request in XXXDomain APIs are not domainGetter, so hard to reuse. But I make it a function to some some lines.